perceval/backends/mozilla/mozillaclub.py

```diff
@@ -0,0 +1,397 @@
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+
+        logger.info("Retrieving cached events: '%s'", self.url)
+
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = next(self.cache.retrieve())
+        sheet_json = json.loads(cache_items)
+
+        parser = MozillaClubParser()
+
+        nevents = 0
+
+        for event in parser.parse(sheet_json):
+            yield event
+            nevents += 1
+
+        logger.info("Total number of events from cache: %i", nevents)
+
+    @classmethod
+    def has_caching(cls):
+        """Returns whether it supports caching items on the fetch process.
+
+        :returns: this backend supports items cache
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an event item."""
+
+        try:
+            a = item.keys()
+        except AttributeError:
+            print ("->"+item+"<-")
+            raise
+        return str(item['Date of Event']+"_"+item['Club Name'])
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a item.
+
+        This backend only generates one type of item which is
+        'event'.
+        """
+        return 'event'
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MozillaClub item.
+
+        The timestamp is extracted from 'end' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        date = str_to_datetime(item['Date of Event'])
+        return float(date.timestamp())
+
+
+class MozillaClubClient:
+    """MozillaClub API client.
+
+    This class implements a simple client to retrieve events from
+    projects in a MozillaClub site.
+
+    :param url: URL of MozillaClub
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, url):
+        self.url = url
+
+    def call(self, uri):
+        """Run an API command.
+
+        :param params: dict with the HTTP parameters needed to run
+            the given command
+        """
+        logger.debug("MozillaClub client calls API: %s", self.url)
+
+        req = requests.get(uri)
+        req.raise_for_status()
+
+        return req.text
+
+    def get_cells(self):
+        """Retrieve all cells from the spreadsheet."""
+
+        logging.info("Retrieving all cells spreadsheet data ...")
+        raw_cells = self.call(self.url)
+
+        return raw_cells
+
+
+class MozillaClubParser:
+    """Git log parser.
```

Comment:
I would like to have in the documentation some examples about how the format of the input is. It is not enough clear how this is parsed and since parsing is a little tricky, it would be useful to have some examples here. Please check Git parser to find good examples.